Add Ramp

eliblock opened the pull request.

eliblock committed:
Add Ramp

Carlgo11 commented on entries/a/app.ramp.com.json:
```diff
@@ -0,0 +1,12 @@
+{
+  "Ramp": {
+    "domain": "app.ramp.com",
+    "additional-domains": ["ramp.com", "demo.ramp.com"],
```
The domain should be without a subdomain.
```suggestion
    "domain": "ramp.com",
```

eliblock replied:
The domains with actual login screens which use MFA are `app.ramp.com` and `demo.ramp.com`. Is it still appropriate to use just the top level domain? Will adjust (and rename the file) if so!

Carlgo11 replied:
Yes we don't list subdomains. Third party apps using our data will look for the top level domain and not the full domain.

Carlgo11 commented on img/a/app.ramp.com.svg:
```diff
@@ -0,0 +1 @@
+<svg width="32" height="32" viewBox="0 0 32 32" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M0 16C0 7.16344 7.16344 0 16 0C24.8366 0 32 7.16344 32 16C32 24.8366 24.8366 32 16 32C7.16344 32 0 24.8366 0 16Z" fill="#F5FF78"/><path d="M24.6155 22.0846V22.1497L15.231 22.153V22.0846C16.5842 21.3127 17.5183 20.5264 18.3588 19.7048H22.2123L24.6155 22.0846ZM22.2903 9.74088L19.9116 7.38434H19.8 ...
```
```suggestion
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 32 32"><path fill="#F5FF78" d="M0 16a16 16 0 1 1 32 0 16 16 0 0 1-32 0Z"/><path fill="#000" d="M24.62 22.08v.07h-9.39v-.07a14.7 14.7 0 0 0 3.13-2.38h3.85l2.4 2.38ZM22.29 9.74l-2.38-2.36h-.07s.04 4.4-3.95 8.39a13.11 13.11 0 0 1-8.5 3.92v.07l2.42 2.4s4.53.04 8.53-3.92c3.98-3.95 3.95-8.5 3.95-8.5Z"/></svg>
```

Carlgo11 commented on entries/a/app.ramp.com.json:
```diff
@@ -0,0 +1,12 @@
+  "Ramp": {
+    "domain": "app.ramp.com",
+    "additional-domains": ["ramp.com", "demo.ramp.com"],
+    "tfa": ["sms", "totp"],
```
If the entry adheres to the code styling that's used on all other entries it's much easier to read and compare.
```suggestion
    "tfa": [
      "sms",
      "totp"
    ],
```

Carlgo11 commented on entries/a/app.ramp.com.json:
```diff
@@ -0,0 +1,12 @@
+    "domain": "app.ramp.com",
+    "additional-domains": ["ramp.com", "demo.ramp.com"],
+    "tfa": ["sms", "totp"],
+    "documentation": "https://support.ramp.com/hc/en-us/articles/5775209997715-Multi-Factor-Authentication",
```
```suggestion
    "documentation": "https://support.ramp.com/hc/en-us/articles/5775209997715",
```

Carlgo11 reviewed (changes requested):
I don't see a region array. Are you sure Ramp is available globally?

eliblock committed:
Apply suggestions from code review

Co-authored-by: Carlgo11 <Carlgo11@users.noreply.github.com>

eliblock committed:
rename files

eliblock committed:
restrict to `us`

Though Ramp is globally accessible, its primary users are US-based
businesses, so adding the region restriction.

Carlgo11 reviewed (approved):
Thanks! :smile:

phallobst commented on entries/r/ramp.com.json:
```diff
@@ -0,0 +1,17 @@
+  "Ramp": {
+    "domain": "ramp.com",
+    "tfa": [
+      "sms",
```
According to the doc page, 2FA via email is also available.
```suggestion
      "sms",
      "email",
```

eliblock replied:
From using the service, email is available only when neither sms nor an authenticator are used (reflected on the docs page as well). That said, adding per your suggestion.

phallobst commented on entries/r/ramp.com.json:
```diff
@@ -0,0 +1,17 @@
+    ],
+    "documentation": "https://support.ramp.com/hc/en-us/articles/5775209997715",
+    "keywords": [
+      "payments",
```
The other entries in the `payments` category are actual payment platforms or services, so this one does not really fit in there, I guess.
```suggestion
```

eliblock committed:
remove `"payments"` keyword

Co-authored-by: Michael Loßin <github@michael-lossin.de>

eliblock committed:
add `email`

Co-authored-by: Michael Loßin <github@michael-lossin.de>
